build_runner/lib/src/build_script_generate/bootstrap.dart

```diff
@@ -18,13 +18,26 @@ final _logger = Logger('Bootstrap');
 
 /// Generates the build script, snapshots it if needed, and runs it.
 ///
+/// The optional [generationOptions] can be used to customize the generated
+/// build script.
+/// The optional [printUncaughtError] function will be invoked when the
+/// build script threw an uncaught error. The default behavior asks users to
+/// open an issue at `dart-lang/build`, this can be customized.
+///
 /// Will retry once on [IsolateSpawnException]s to handle SDK updates.
 ///
 /// Returns the exit code from running the build script.
 ///
 /// If an exit code of 75 is returned, this function should be re-ran.
-Future<int> generateAndRun(List<String> args, {Logger logger}) async {
+Future<int> generateAndRun(
+  List<String> args, {
+  Logger logger,
+  BuildScriptGenerationOptions generationOptions =
+      const BuildScriptGenerationOptions(),
+  void Function(Stdout output) printUncaughtError,
```

Comment:
This is a bit of a weird api - really it just looks like it is allowing you to customize the message?

Should we change it to `void Function(Object error, StackTrace stackTrace) handleUncaughtError` which would be the more typical way of allowing users to handle uncaught errors?

I would probably implement that like:

```dart
handleUncaughtError ??= (e, _) {
  printUncaughtError(stderr);
  stderr..writeln(error)..writeln(Trace.parse(trace).terse);
};
...
errorListener = errorPort.listen((e) {
  final error = e[0];
  final trace = Trace.parse(e[1] as String).terse;
  handleUncaughtError(error, trace);
  if (scriptExitCode == 0) scriptExitCode = 1;
});
```

Possibly also allowing a function with a single error argument but im not sure its worth the complexity and loss of static validation on the function type.

Reply:
> Should we change it to `void Function(Object error, StackTrace stackTrace) handleUncaughtError` which would be the more typical way of allowing users to handle uncaught errors?

Yes, definitely. I just wanted to be able to remove the "Report this at dart-lang/build" message and missed the obvious way.